pkg/nuctl/test/get_test.go

```diff
@@ -45,6 +45,103 @@ func (suite *GetTestSuite) SetupSuite() {
 	})
 }
 
+func unduplicate(array []string) []string {
+	unduplicatedArray := []string{}
+
+	// iterate over array to find duplicated values
+	for _, value := range array {
+		add := true
+		for _, returnValue := range array {
+			if returnValue == value {
+				add = false
+			}
+		}
+		if add {
+			unduplicatedArray = append(unduplicatedArray, value)
+		}
+	}
+	return unduplicatedArray
+}
+
+func (suite *GetTestSuite) TestMultipleGet() {
+	numOfFunctions := 4
+	var functionNames []string
+
+	// initialize functions
+	for functionIdx := 0; functionIdx < numOfFunctions; functionIdx++ {
+		uniqueSuffix := fmt.Sprintf("-%s-%d", xid.New().String(), functionIdx)
+		imageName := "nuclio/deploy-test" + uniqueSuffix
+		functionName := "reverser" + uniqueSuffix
+
+		// add function name to list
+		functionNames = append(functionNames, functionName)
+		namedArgs := map[string]string{
+			"path":    path.Join(suite.GetFunctionsDir(), "common", "reverser", "golang"),
+			"image":   imageName,
+			"runtime": "golang",
+			"handler": "main:Reverse",
+		}
+
+		err := suite.ExecuteNutcl([]string{
+			"deploy",
+			functionName,
+			"--verbose",
+			"--no-pull",
+		}, namedArgs)
+		suite.Require().NoError(err)
+
+		// cleanup
+		defer func() {
+
+			// make sure to clean up after the test
+			suite.dockerClient.RemoveImage(imageName)
+		}()
+	}
+
+	// number of combinations need to be check
+	tests := [][]string{{functionNames[0], functionNames[1], functionNames[2]}, {functionNames[1], functionNames[1]},
+		{functionNames[0], functionNames[2], functionNames[2]}}
+
+	// Iterate over tests and check for right results for each one
+	for _, test := range tests {
+		err := suite.ExecuteNutcl(append([]string{"get", "function"}, test...), nil)
+		suite.Require().NoError(err)
+		foundFunctions := make([]int, len(unduplicate(test)))
+
+		// iterate over all lines in get result. for each function created in this test that we find,
+		// set the equivalent boolean in foundFunctions
+		scanner := bufio.NewScanner(&suite.outputBuffer)
+
+		for scanner.Scan() {
+			for functionIdx, functionName := range unduplicate(test) {
+
+				// if the function name is in the list, remove it
+				if strings.Contains(scanner.Text(), functionName) {
+
+					// increase times that function has been found
+					foundFunctions[functionIdx]++
+					break
+				}
+			}
+		}
+
+		for _, foundFunction := range foundFunctions {
+			suite.Require().Equal(1, foundFunction)
+		}
+
+		// reset suite's outputBuffer
+		suite.outputBuffer.Reset()
+	}
+
+	// use nuctl to delete the functions when we're done
+	err := suite.ExecuteNutcl(append([]string{"delete", "fu"}, functionNames...), nil)
+	suite.Require().NoError(err)
+
+	// use nuctl delete --all flag to delete rest of the functions
```

Comment:
What rest of the functions? We should've deleted all of them with the above delete call